pywbem/_server.py

```diff
@@ -1122,3 +1320,65 @@ def _determine_profiles(self):
         mp_insts = self._conn.EnumerateInstances("CIM_RegisteredProfile",
                                                  namespace=self.interop_ns)
         self._profiles = mp_insts
+
+    def _count_associators(self):
+        """
+        Create count of associators for CIM_ReferencedProfile using the
+        antecedent and dependent reference properties as ResultRole for each
+        profile defined in server.profiles and return a dictionary of the
+        count. This code does a shortcut in executing EnumerateInstances to get
+        CIM_ReferencedProfile and processing the association locally.
+
+        Returns
+          Dictionary where the keys are the profile names and the value for
+          each key is a dictionary with two keys ('dep' and 'ant')
+          The value for each key is the count of associator paths when the key
+          value is used as the ResultRole
+
+        """
+        try:
+            ref_insts = self.conn.EnumerateInstances(
+                "CIM_ReferencedProfile", namespace=self.interop_ns)
+            # Remove host from responses since the host causes confusion
+            # with the enum of registered profile. Enums do not return host but
+            # the associator properties contain host
+            for ref_inst in ref_insts:
+                for prop in ref_inst.values():
+                    prop.host = None
+
+        except Error as er:
+            print('CIM_ReferencedProfile failed for server=%s exception=%s'
```

Comment:
What about that class failed?

Reply:
I am uncomfortable with this one.  Possibilities are:

1. No try block, just let the error flow upward if CIMReferencedProfile enumerate results in exception. That means that there is no possibility of scoping information.

2. Use try block and try to pick out the types of error. Generally it could be 
    a. Class failed (that means no concept of scoping, etc.
    b. Enumeration returns NOTSUPPORTED. means that enumeration of this association is not supported.
    c. Other errors. Total failure somewhere.

However, the result is:
    a. Any error other than NOTSUPPORTED. complete failure
    b. NOTSUPPRORTED - we could go back to determining from executing associations which takes 10+++ times as long. (minutes rather than seconds)

NOTE: All the tested servers do return enumeration.

Reply:
DISCUSSION ITEM

Reply:
See above.